dtls.c

```diff
@@ -4282,6 +4283,17 @@ dtls_handle_message(dtls_context_t *ctx,
                  content_type, epoch, pkt_seq_nr, rlen);
     }
 
+    if (!peer) {
```

Comment:
Since re-validation is triggered always at the end of this while-loop, we could decide to move the initial retrieval into this loop.

Reply:
I kept the copy for the different logging message. 
That could be also achieved, when the snippets are merged.
I will try it.